### ISSUE TITLE
Pyright doesn't accept a union type (e.g. `int | str`) for isinstance calls

### ISSUE THREAD

**Issue by StockerMC** [microsoft/pyright#2565]
**Describe the bug**
Pyright doesn't accept union types for the type of isinstance calls.

**Expected behavior**
The following code to successfully type check.

**Screenshots or Code**
```python
isinstance(1, int | str)
```
```
error: Argument of type "Type[int] | Type[str]" cannot be assigned to parameter "__class_or_tuple" of type "type | UnionType | Tuple[type | UnionType | Tuple[Any, ...], ...]" in function "isinstance"
    Type "object" cannot be assigned to type "type | UnionType | Tuple[type | UnionType | Tuple[Any, ...], ...]"
      "object" is incompatible with "type"
      "object" is incompatible with "UnionType"
      "object" is incompatible with "Tuple[type | UnionType | Tuple[Any, ...], ...]"
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line (pyright 1.1.187)

**Comment by erictraut** [comment]
Oh interesting, I didn't realize that Python 3.10 added support for this. All previous versions of Python generated runtime errors for `isinstance(1, Union[int, str])`. Python 3.10 appears to accept both the old `Union` syntax and the new `|` syntax in this context. I presume that it's the equivalent of `isinstance(1, (int, str))`?

**Comment by JelleZijlstra** [comment]
It's in https://www.python.org/dev/peps/pep-0604/#isinstance-and-issubclass though it's not specified very precisely. But yes, the equivalence is as you say.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.189, which I just published. It will also be included in the next release of pylance.
